libpermian/plugins/anaconda_webui/hypervisor.py

```diff
@@ -0,0 +1,140 @@
+import subprocess
+import time
+import logging
+import os
+import tempfile
+import random
+
+
+START_PORT = '42000'
+LOGGER = logging.getLogger(__name__)
+
+
+class Hypervisor():
+    def __init__(self, host):
+        self.host = host
+
+        if host in ['localhost', '127.0.0.1', '::1']:
+            self.remote = False
+            self.qemu_host = 'qemu:///system'
+        else:
+            self.remote = True
+            self.qemu_host = f'qemu+ssh://{host}/system'
+            self.tempdir = tempfile.TemporaryDirectory(prefix='permian_sshctl.')
+            self.ssh_ctl_file = os.path.join(self.tempdir.name, '%r@%h:%p')
+            self._open_connection()
+
+    def setup(self):
+        # Try to create storage pool boot-scratch to avoid issues later, if it exist its ok for this command to fail.
+        self.virsh_call([
+            'pool-create-as', '--name', 'boot-scratch',
+            '--type', 'dir', '--target' ,'/var/lib/libvirt/boot',
+            '--build'
+        ])
+
+    def forwarding_cleanup(self):
+        self._close_conntection()
+        self.tempdir.cleanup()
+
+    def _open_connection(self):
```

Comment:
It would be good if `_open_connection` could be called repeatedly not doing anything if connection has been already established. This way, the method could be called "on demand" before any use of the ssh connection basically just making sure the connection is active and if not activating it so that it can be used in the code. (Effective allowing it to be simply used in the `_run_cmd` method.)

Reply:
I don't see a reason to call this method multiple times, its called once on init